File: Folder-1/two sum.py
#Solution 1, Brute force

def two_sum(array, target):
    for i in range(len(array)-1):
        for j in range(i+1,len(array)):
            if array[i]+array[j] == target:
                return [i,j]
    return []
            
            
array1 = [1,2,3,4,5]
target = 9

print(two_sum(array1,target))

#Solution 2
#Time Complexity O(n), Space Complexity O(n)
def two_sum(array,target):
    num_available = {}
    for i in range(len(array)):
        needed_val = target - array[i]
        if needed_val in num_available:
            return [i,num_available[needed_val]]
        else:
            num_available[array[i]]=i
    return []  

File: Folder-1/two_sum.js
function findIndicesSum(array, targetValue) {
    for (let i = 0; i < array.length - 1; i++) {
        for (let j = i + 1; j < array.length; j++) {
            if (targetValue === array[i] + array[j]) {
                return [i, j];
            }
        }
    }
    return [];
}

array1 = [1, 2, 3, 4, 5]
target = 9

console.log(findIndicesSum(array1, target))